### PR TITLE
[Integration][Wiz] Add configurable limit for wiz issues

### PR DESCRIPTION
**Pull request by phalbert**
# Description

Currently, the Wiz import feature only displays the top 250 recent issues due to a pre-set limit. Make this limit configurable by using a selector `maxPages`

## Type of change

Please leave one option from the following and delete the rest:

- [ ] New feature (non-breaking change which adds functionality)

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/wiz/.port/resources/port-app-config.yaml [hunk 1]
```diff
@@ -16,7 +16,9 @@ resources:
             description: .description
   - kind: issue
     selector:
-      query: 'true'
+      query: "true"
+      maxPages: 20
```
Comment (PeyGis) [inline integrations/wiz/.port/resources/port-app-config.yaml:20]: I'm curious why the default is 20. Is this the maximum number of pagination request that can be made or the number of issues that can be returned?

File: integrations/wiz/.port/resources/port-app-config.yaml [hunk 2]
```diff
@@ -16,7 +16,9 @@ resources:
...
+      statusList: ["OPEN", "IN_PROGRESS"]
```
Comment (PeyGis) [inline integrations/wiz/.port/resources/port-app-config.yaml:21]: Can we rename it to statuses?

File: integrations/wiz/main.py [hunk 3]
```diff
@@ -26,31 +27,53 @@ def init_client() -> WizClient:
...
+    """Resync projects from Wiz."""
+    wiz_client = init_client()
+    async for projects in wiz_client.get_projects():
+        logger.info(f"Received {len(projects)} projects")
+        yield projects
+
+
+@ocean.on_resync(kind=ObjectKind.ISSUE)
+async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    logger.info(f"Resyncing {kind} kind")
```
Comment (PeyGis) [inline integrations/wiz/main.py:41]: You can remove this since the ocean framework logs the resync events

File: integrations/wiz/main.py [hunk 4]
```diff
@@ -26,31 +27,53 @@ def init_client() -> WizClient:
...
+    wiz_client = init_client()
+
+    max_pages = cast(IssueResourceConfig, event.resource_config).selector.max_pages
+    status_list = cast(IssueResourceConfig, event.resource_config).selector.status_list
```
Comment (PeyGis) [inline integrations/wiz/main.py:45]: I would change this to something like

selector = cast(IssueResourceConfig, event.resource_config).selector

then you can access it directly in the async for as selector.max_pages and selector.statuses

File: integrations/wiz/overrides.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    ResourceConfig,
+    PortAppConfig,
+    Selector,
+)
+from pydantic import Field
+
+
+class IssueSelector(Selector):
+    max_pages: int = Field(alias="maxPages", default=20)
+    status_list: list[str] = Field(alias="statusList", default_factory=list)
```
Comment (PeyGis) [inline integrations/wiz/overrides.py:13]: remember to rename this to statuses based on previous comment

**Review by PeyGis — CHANGES_REQUESTED** [review]
Looks good. Just some few comments

File: integrations/wiz/.port/resources/port-app-config.yaml (hunk 1 above)
Comment (phalbert) [reply integrations/wiz/.port/resources/port-app-config.yaml:20]: This is the number of pages. Each page fetch returns 100 items. So by default 2000 items will be returned

Comment (PeyGis) [reply integrations/wiz/.port/resources/port-app-config.yaml:20]: Gotcha. For other integrations that brings in issues such as Snyk, SonarQube, Sentry etc, it is possible to ingest any number of entities without limitation. is the 2000 the limit recommended by Wiz or something that was agreed with the customers who are using the integration?

File: integrations/wiz/.port/resources/port-app-config.yaml (hunk 2 above)
Comment (phalbert) [reply integrations/wiz/.port/resources/port-app-config.yaml:21]: Let's let it be. statuses is less easy to enunciate when spoken. I prefer this.

File: integrations/wiz/CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 <!-- towncrier release notes start -->
+# Port_Ocean 0.1.39 (2024-07-25)
+
+### Improvements
+
+- Add paramter for setting number of issues to fetch
```
Comment (yairsimantov20) [inline integrations/wiz/CHANGELOG.md:13]: "Add" there for its a new feature and not an improvement

File: integrations/wiz/main.py [hunk 7]
```diff
@@ -26,31 +27,50 @@ def init_client() -> WizClient:
     )
 
 
-@ocean.on_resync()
-async def on_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+@ocean.on_resync(kind=ObjectKind.PROJECT)
+async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """Resync projects from Wiz."""
```
Comment (yairsimantov20) [inline integrations/wiz/main.py:32]: Remove

File: integrations/wiz/main.py [hunk 8]
```diff
@@ -26,31 +27,50 @@ def init_client() -> WizClient:
...
+    async for _issues in wiz_client.get_issues():
+        yield [
+            issue["sourceRule"]
+            for issue in _issues
+            if issue.get("sourceRule") is not None
+        ]
+
+
+@ocean.on_resync(kind=ObjectKind.SERVICE_TICKET)
+async def on_resync_tickets(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
```
Comment (matan84) [inline integrations/wiz/main.py:66]:
```suggestion
async def resync_tickets(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
```

File: integrations/wiz/main.py [hunk 9]
```diff
@@ -26,31 +27,50 @@ def init_client() -> WizClient:
...
+    async for _issues in wiz_client.get_issues(
+        max_pages=selector.max_pages, 
+        status_list=selector.status_list
+    ):
+        logger.info(f"Received {len(_issues)} issues")
+        yield _issues
+
+
+@ocean.on_resync(kind=ObjectKind.CONTROL)
+async def on_resync_controls(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
```
Comment (matan84) [inline integrations/wiz/main.py:54]:
```suggestion
async def resync_controls(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
```

File: integrations/wiz/main.py [hunk 10]
```diff
@@ -26,31 +27,50 @@ def init_client() -> WizClient:
...
+async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """Resync projects from Wiz."""
+    wiz_client = init_client()
+    async for projects in wiz_client.get_projects():
+        logger.info(f"Received {len(projects)} projects")
+        yield projects
+
+
+@ocean.on_resync(kind=ObjectKind.ISSUE)
+async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
```
Comment (matan84) [inline integrations/wiz/main.py:40]:
```suggestion
async def resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
```

**Review by matan84 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**